scripts/dev/e2e.py

```diff
@@ -207,32 +220,65 @@ def parse_args():
         type=bool,
         default=False,
     )
+    parser.add_argument(
+        "--skip-image-build", help="Skip building images", type=bool, default=False,
+    )
+    parser.add_argument(
+        "--tag",
+        help="Tag for the images, it will be the same for all images",
+        type=str,
+        default="latest",
+    )
+    parser.add_argument(
+        "--dump_diagnostic",
+        help="Dump diagnostic information into files",
+        type=bool,
+        default=False,
+    )
+    parser.add_argument("--config_file", help="Path to the config file")
     return parser.parse_args()
 
 
 def main():
```

Comment:
our main function is getting a big large. Maybe we could split some of these things up into smaller functions. (e.g. building & pushing images)